File: Lab-5/reversing_words4.cpp
#include <vector>
#include <string>
#include <iostream>
using namespace std;

string reverse_input( signed int n, string prev ){
    string new_input = "";
    cin >> new_input;

    if (n == 0) return( prev );
    else{
        if ( n < 0 ){
            reverse_input( stoi(new_input) , "");
            return( prev );
        }
        else{
            cout << reverse_input( n - 1 , new_input ) << endl;
            return( prev );
        }
    }
}

int main(){
    reverse_input( -1 , "");
    return(0);
}

File: Lab-5/reversing_words4.py
from easyinput import read_line,read
def reverse_input(n, prev):
    new_input = read_line()                   
    if n == 0:
        return ( prev ) 
    
    else:
        if n == -1 :
            reverse_input( new_input , "")
            return( prev )
        else:
            print (reverse_input( n - 1, new_input ))
            return( prev )

reverse_input("",read(int))